internals/overlord/state/notices.go

```diff
@@ -81,7 +85,21 @@ type Notice struct {
 }
 
 func (n *Notice) String() string {
-	return fmt.Sprintf("Notice %s (%s:%s)", n.id, n.noticeType, n.key)
+	return fmt.Sprintf("Notice %s (%d:%s:%s)", n.id, n.userID, n.noticeType, n.key)
+}
+
+// UserID returns whether the notice has a user ID, and if so, what the UID is.
+func (n *Notice) UserID() (hasUserID bool, userID uint32) {
+	// Importantly, doesn't expose the address of notice's user ID, so the
+	// value cannot be mutated.
```

Comment:
```suggestion
	// caller cannot mutate the value.
```

Reply:
I personally don't think it's terrible to just return the `*uint32` directly. The caller (part of the Pebble project) would have to be kinda malicious to do that. No strong feelings, though.

Reply:
I prefer not to expose it, and additionally this helps the caller make sure that they're comparing the value of the user ID, not its address, if they compare it to some other `*uint32` on their end. The type checking will catch that their `*uint32` needs to be dereferenced.